.circleci/config.yml

```diff
@@ -1,582 +1,86 @@
-      - persist_to_workspace:
-          root: .
-          paths:
-            - ./dist/*.whl
-            - ./dist/*.tar.gz
-
-  publish-snapshot-python:
-    working_directory: ~/openlineage
-    docker:
-      - image: circleci/python:3.6
-    steps:
-      - *checkout_project_root
-      - attach_workspace:
-          at: .
-      - run: pip install wheel twine
-      - run: python -m twine upload --non-interactive --verbose -u $ARTIFACTORY_USERNAME -p $ARTIFACTORY_PASSWORD --repository-url https://datakin.jfrog.io/artifactory/api/pypi/pypi-public-libs-release dist/*
-
-  release-python:
-    working_directory: ~/openlineage
-    docker:
-      - image: circleci/python:3.6
-    steps:
-      - *checkout_project_root
-      - attach_workspace:
-          at: .
-      - run: pip install wheel twine
-      - run: python -m twine upload --non-interactive --verbose --repository pypi dist/*
-
-  publish-spec:
-    working_directory: ~/openlineage
-    docker:
-      - image: cimg/base:2021.07
-    steps:
-      - *checkout_project_root
-      - add_ssh_keys:
-          fingerprints:
-            - "1c:d1:da:e8:76:d7:f7:04:31:07:18:fd:55:ca:e1:2e"
-      - run: spec/release.sh
-
-  build-proxy-backend:
-    working_directory: ~/openlineage/proxy
-    docker:
-      - image: cimg/openjdk:11.0
-    steps:
-      - *checkout_project_root
-      - restore_cache:
-          keys:
-            - v1-proxy-{{ .Branch }}-{{ .Revision }}
-            - v1-proxy-{{ .Branch }}
-      - run: ./gradlew --no-daemon --stacktrace build
-      - run: ./gradlew --no-daemon jacocoTestReport
-      - run: bash <(curl -s https://codecov.io/bash)
-      - store_test_results:
-          path: proxy/build/test-results/test
-      - store_artifacts:
-          path: build/reports/tests/test
-          destination: test-report
-      - save_cache:
-          key: v1-proxy-{{ .Branch }}-{{ .Revision }}
-          paths:
-            - ~/.gradle
-
-  release-proxy-backend:
-    working_directory: ~/openlineage/proxy
-    docker:
-      - image: cimg/openjdk:11.0
-    steps:
-      - *checkout_project_root
+          name: Install yq
+          command: go install "github.com/mikefarah/yq/v4@latest"
       - run: |
-          # Get, then decode the GPG private key used to sign *.jar
-          export ORG_GRADLE_PROJECT_signingKey=$(echo $GPG_SIGNING_KEY | base64 -d)
-          export RELEASE_PASSWORD=$(echo $SONATYPE_PASSWORD)
-          export RELEASE_USERNAME=$(echo $SONATYPE_USER)
-
-          # publish jar to maven local so it can be found by dependents
-          ./gradlew publishToMavenLocal
-
-          # Publish *.jar
-          ./gradlew publish
+          # Based on the changed files (or if we are on the main branch), generate a yaml file with
+          # a list of workflow files. Use those files to build a workflow with a union of all tasks
+
+          function check_change () {
```

Comment:
Very cool! But, just curious, since we pull in the `circleci/path-filtering` orb, why not use the `path-filtering/filter` job that takes care of the filtering for us based? (Use the example [config.yml](https://circleci.com/docs/2.0/configuration-cookbook/?section=examples-and-guides#execute-specific-workflows-or-steps-based-on-which-files-are-modified) for dynamic configuration as a reference)

Reply:
Comments added